### PR TITLE
feat(api): Change alert rule creation to be at the org level(SEN-989)

### PR DESCRIPTION
**Pull request by wedamija**
Change alert rule creation to be at the org level, and allow mutiple projects to be subscribed to
an alert rule. Add `bulk_create_snuba_subscriptions` so that we can create all the subscriptions
efficiently. Currently doesn't batch but we'll add this in a future pr.

Update the alert rule write serializer to also accept an organization. We'll change it to accept
projects as params later, for now still passing as context.

Remove deprecated columns from:
- AlertRule
  - subscription_id (handled in snuba QuerySubscription now)
   - aggregations (moved to singular column)
   - query_subscription (replaced my m2m table)
   - project (replaced by QuerySubscription)
- QuerySubscription
  - aggregations (replaced by singular column)

Will delete these from the database in a later pr, we need to drop the columns from Django first.

### CODE REVIEW

File: src/sentry/incidents/endpoints/serializers.py [hunk 1]
```diff
@@ -93,8 +97,6 @@ def _remove_unchanged_fields(self, instance, validated_data):
             # Remove any fields that haven't actually changed
             if isinstance(value, Enum):
                 value = value.value
-            elif field_name == "aggregations":
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/serializers.py]: This is unused due to a previous change, just removing

File: src/sentry/incidents/logic.py [hunk 2]
```diff
@@ -618,9 +611,20 @@ def create_alert_rule(
             resolve_threshold=resolve_threshold,
             threshold_period=threshold_period,
         )
-        AlertRuleQuerySubscription.objects.create(
-            query_subscription=query_subscription, alert_rule=alert_rule
+        subscriptions = bulk_create_snuba_subscriptions(
```
Comment (markstory) [inline src/sentry/incidents/logic.py]: If this transaction fails due to races, or other database problems should we attempt to clean up the subscriptions in snuba?

File: src/sentry/incidents/models.py [hunk 3]
```diff
@@ -291,12 +291,6 @@ class AlertRule(Model):
     threshold_period = models.IntegerField()
     date_modified = models.DateTimeField(default=timezone.now)
     date_added = models.DateTimeField(default=timezone.now)
-    # These will be removed after we've made these columns nullable. Moving to
-    # QuerySubscription
-    subscription_id = models.UUIDField(db_index=True, null=True)
-    aggregations = ArrayField(of=models.IntegerField)
-    query_subscription = FlexibleForeignKey("sentry.QuerySubscription", unique=True, null=True)
-    project = FlexibleForeignKey("sentry.Project", db_index=False, db_constraint=False, null=True)
```
Comment (markstory) [inline src/sentry/incidents/models.py]: :clap:

File: src/sentry/incidents/logic.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/incidents/logic.py]: We'll handle all this retry/cleanup logic in the tasks. For now I don't care too much if things end up in a bad state (Snuba isn't even writing data to kafka at this point)

**Review by markstory — APPROVED** [review]

**Pull request merged**